oemof/solph/constraints.py

```diff
@@ -60,38 +78,41 @@ def emission_limit(om, flows=None, limit=None):
     flows : dict
         Dictionary holding the flows that should be considered in constraint.
         Keys are (source, target) objects of the Flow. If no dictionary is
-        given all flows containing the 'emission_factor' attribute will be
+        given all flows containing the keyword attribute will be
         used.
+    keyword : attribute to consider
     limit : numeric
-        Absolute emission limit for the energy system.
+        Absolute limit of keyword attribute for the energy system.
 
     Note
     ----
-    Flow objects required an emission attribute!
+    Flow objects required an attribute named like keyword!
 
     """
-
     if flows is None:
         flows = {}
         for (i, o) in om.flows:
-            if hasattr(om.flows[i, o], 'emission_factor'):
+            if hasattr(om.flows[i, o], keyword):
                 flows[(i, o)] = om.flows[i, o]
 
     else:
         for (i, o) in flows:
-            if not hasattr(flows[i, o], 'emission_factor'):
+            if not hasattr(flows[i, o], keyword):
                 raise AttributeError(
                     ('Flow with source: {0} and target: {1} '
-                     'has no attribute emission_factor.').format(i.label,
-                                                                 o.label))
+                     'has no attribute {2}.').format(
+                        i.label, o.label, keyword))
+
+    limit_name = "integral_limit_"+keyword
 
-    om.total_emissions =  po.Expression(
+    setattr(om, limit_name, po.Expression(
         expr=sum(om.flow[inflow, outflow, t] * om.timeincrement[t] *
-                 flows[inflow, outflow].emission_factor
+                 sequence(getattr(flows[inflow, outflow], keyword))[t]
```

Comment:
When we allow time dependent weights, this should also be covered in the unit tests.

Reply:
Yeah sure. Do you think it is sufficient to adapt source1 in the constraint test?
```suggestion
                 solph.Source(label='source1', outputs={bel: solph.Flow(
                     nominal_value=100, emission_factor=[0.5, 0.6, 0.7])})
```
If you want, I can adapt the test and lp file...?

Reply:
Done.

Reply:
Done.